config/config.go

```diff
@@ -1126,7 +1128,13 @@ func (c *EC2SDConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
 		return err
 	}
 	if c.Region == "" {
-		return fmt.Errorf("EC2 SD configuration requires a region")
+		sess := session.Must(session.NewSession())
```

Comment:
The errors seems to be caused by wrong configuration.
https://github.com/aws/aws-sdk-go/blob/v1.8.35/aws/session/session.go#L100
https://github.com/aws/aws-sdk-go/blob/v1.8.35/aws/session/session.go#L257
https://github.com/aws/aws-sdk-go/blob/v1.8.35/aws/session/session.go#L335
https://github.com/aws/aws-sdk-go/blob/v1.8.35/aws/session/session.go#L339
https://github.com/aws/aws-sdk-go/blob/v1.8.35/aws/session/session.go#L352

But it is better to add error handling, so I add it :-)